libhb/audio_resample.c

```diff
@@ -85,6 +105,11 @@ void hb_audio_resample_set_channel_layout(hb_audio_resample_t *resample,
             // Dolby Surround is Stereo when it comes to remixing
             channel_layout = AV_CH_LAYOUT_STEREO;
         }
+        if (resample->out.channel_layout == AV_CH_LAYOUT_MONO &&
+            is_mono(channel_layout))
+        {
+            resample->out.channel_layout = channel_layout;
+        }
         resample->in.channel_layout = channel_layout;
     }
```

Comment:
I posted this as a PR because I expected you would have valuable input, thanks.  Both of your suggestions added.